src/Og.php

```diff
@@ -169,6 +171,31 @@ public static function getEntityGroups($entity_type, $entity_id, $states = [OG_S
   }
 
   /**
+   * Return TRUE if entity belongs to a group.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $group
+   *   The group entity to check.
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to get groups for.
+   * @param array $states
+   *   (optional) Array with the membership states to check the membership.
+   *   Defaults to active memberships.
+   *
+   * @return bool
+   *   TRUE if the entity (e.g. the user) belongs to a group and is not pending
+   *   or blocked.
+   */
+  public static function isMember(EntityInterface $group, EntityInterface $entity, $states = [OG_STATE_ACTIVE]) {
+    $groups = static::getEntityGroups($entity, $states);
+    $group_entity_type_id = $group->getEntityTypeId();
+    // We need to create a map of the group ids as Og::getEntityGroups returns a
+    // map of membership_id => group entity for each type.
+    return !empty($groups[$group_entity_type_id]) && in_array($group->id(), array_map(function($group_entity) {
```

Comment:
Alright, didn't knew about that, and commends are hard to read.

Ship it I guess?

On Mon, Dec 7, 2015 at 8:08 PM, Damian Lee <notifications@github.com> wrote:

> In src/Og.php
> <https://github.com/amitaibu/og/pull/53#discussion_r46861821>:
>
> > +   * @param \Drupal\Core\Entity\EntityInterface $entity
> > +   *   The entity to get groups for.
> > +   * @param array $states
> > +   *   (optional) Array with the membership states to check the membership.
> > +   *   Defaults to active memberships.
> > +   *
> > +   * @return bool
> > +   *   TRUE if the entity (e.g. the user) belongs to a group and is not pending
> > +   *   or blocked.
> > +   */
> > +  public static function isMember(EntityInterface $group, EntityInterface $entity, $states = [OG_STATE_ACTIVE]) {
> > +    $groups = static::getEntityGroups($entity, $states);
> > +    $group_entity_type_id = $group->getEntityTypeId();
> > +    // We need to create a map of the group ids as Og::getEntityGroups returns a
> > +    // map of membership_id => group entity for each type.
> > +    return !empty($groups[$group_entity_type_id]) && in_array($group->id(), array_map(function($group_entity) {
>
> Otherwise, of course!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/amitaibu/og/pull/53/files#r46861821>.
>